pyogrio/_io.pyx

```diff
@@ -626,17 +711,24 @@ def ogr_read(
     ogr_dataset = ogr_open(path_c, 0, kwargs)
     ogr_layer = get_ogr_layer(ogr_dataset, layer)
 
-    # Apply the attribute filter
-    if where is not None and where != "":
-        apply_where_filter(ogr_layer, where)
-
+    if fids is not None:
+        if where is not None or bbox is not None or skip_features or max_features:
+            raise ValueError(
+                "cannot set both 'fids' and one of 'where', 'bbox', "
+                "'skip_features' or 'max_features'"
+            )
+        fids = np.asarray(fids, dtype=np.intc)
+    else:
```

Comment:
Suggestion: move the substantive work here (e.g., applying where filter, etc) down to the `else` block on 755, so that this conditional test up top is just for validating parameters.